src/Symfony/Component/JsonPath/Test/JsonPathAssertionsTrait.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\JsonPath\Test;
+
+use Symfony\Component\JsonPath\JsonCrawler;
+use Symfony\Component\JsonPath\JsonPath;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ *
+ * @experimental
+ */
+trait JsonPathAssertionsTrait
+{
+    public static function assertJsonPathEquals(mixed $expectedValue, JsonPath|string $jsonPath, string $json, string $message = ''): void
+    {
+        self::assertEquals($expectedValue, self::evaluateJsonPath($jsonPath, $json), $message);
+    }
+
+    public static function assertJsonPathNotEquals(mixed $expectedValue, JsonPath|string $jsonPath, string $json, string $message = ''): void
+    {
+        self::assertNotEquals($expectedValue, self::evaluateJsonPath($jsonPath, $json), $message);
+    }
+
+    public static function assertJsonPathCount(int $expectedCount, JsonPath|string $jsonPath, string $json, string $message = ''): void
+    {
+        $actualCount = \count((new JsonCrawler($json))->find($jsonPath));
+
+        self::assertEquals($expectedCount, $actualCount, $message);
+    }
+
+    public static function assertJsonPathSame(mixed $expectedValue, JsonPath|string $jsonPath, string $json, string $message = ''): void
+    {
+        self::assertSame($expectedValue, self::evaluateJsonPath($jsonPath, $json), $message);
```

Comment:
should we provide those as constraints that can be reused with APIs like `logicalOr()`, as done for browserkit assertions ? The way those assertions are written right now does not make their logic reusable.